W-16981289-VPC-Upgrade (#751)
mulesoft/docs-runtime-manager: cloudhub/modules/ROOT/pages/vpc-upgrade.adoc

= CloudHub VPC to CloudHub 2.0 Private Space Upgrade  

//Overview

> To migrate your workloads from CloudHub to CloudHub 2.0, you need to set up your CloudHub 2.0 infrastructure and network connections. One approach is to create new CloudHub 2.0 private spaces with configurations that differ from your existing CloudHub setup. 

Reviewer: I'm not sure what the second sentence means.

Are you saying:
If you don't need to keep your existing configuration, just create a new CH2 private space.

Author: Actually, what we mean is that one way to migrate to CH2 is by creating CH2 private spaces from scratch, which means that you have to also create new configurations. As opposed to using the in-product VPC upgrade tool, which copies the existing VPC configuration when creating the new CH2 PS.

> However, if you prefer to keep your infrastructure configuration the same, you can use the in-product VPC upgrade tool to easily migrate your CloudHub dedicated Virtual Private Cloud (VPC) to a CloudHub 2.0 private space.

Reviewer: ```suggestion
To retain your infrastructure configuration, use the in-product VPC upgrade tool to easily migrate your CloudHub dedicated Virtual Private Cloud (VPC) to a CloudHub 2.0 private space.
```

> Once your new private space is ready, you can gradually migrate your applications from the CloudHub VPC to the CloudHub 2.0 private space. After all your applications are successfully moved, you can decommission your old VPC.

Reviewer: ```suggestion
When your new private space is ready, you can gradually migrate your applications from the CloudHub VPC to the CloudHub 2.0 private space. After all your applications are successfully moved, you can decommission your old VPC.
```

Reviewer: (Don't use "once" for time:
https://confluence.internal.salesforce.com/display/MTDT/MuleSoft+CX+Writing+Style+Reference#MuleSoftCXWritingStyleReference-%22once%22)

// Benefits

> Using this VPC in-product upgrade tool saves you time by eliminating the need to acquire new CIDR blocks, set up new network connections, and configure the private space from the ground up.

Reviewer: ```suggestion
Using the in-product VPC upgrade tool saves you time because you don't need to acquire new CIDR blocks, set up new network connections, and configure the private space from the ground up.
```

Reviewer: Nice benefits statement!

The tool provisions the infrastructure for a CloudHub 2.0 private space by cloning the existing configurations of your eligible CloudHub dedicated VPC, using the same CIDR block. 

> If your VPC has existing Transit Gateways (TGW) or Virtual Private Networks (VPN) connections, they are seamlessly transferred to the new CloudHub 2.0 private space. This ensures that any new applications deployed into the new private space can communicate effectively.

Reviewer: ```suggestion
If your VPC has existing Transit Gateways (TGW) or Virtual Private Networks (VPN) connections, they transfer seamlessly to the new CloudHub 2.0 private space. This migration ensures that any new applications deployed into the new private space can communicate effectively.
```

> The default Cloudhub certificates are configured as part of the private space's default certificates. The DLB custom domains of the migrated VPC are created as part of the private space's custom TLS certificates.

Reviewer: ```suggestion
The default CloudHub certificates are configured as part of the private space's default certificates. The DLB custom domains of the migrated VPC are created as part of the private space's custom TLS certificates.
```

[NOTE]

> During the VPC upgrade process, you use doubled entitlements for both VPC & Network Connections. Work with your account representative to make sure you stay compliant on entitlements and finish the upgrade on time.

Reviewer: ```suggestion
During the VPC upgrade process, you use doubled entitlements for your VPC and network connections. Work with your account representative to remain compliant with entitlements and finish the upgrade on time.
```

== Understand Eligibility
//Which VPCs are eligible for upgrade

> Due to their architectural differences and distinct interfaces, CloudHub 2.0 doesn't support some features in the same way CloudHub does. Follow this eligibility checklist to confirm which VPCs are eligible for the upgrade:

Reviewer: ```suggestion
Due to their architectural differences and distinct interfaces, CloudHub 2.0 supports some features differently than CloudHub. Follow this eligibility checklist to confirm which VPCs are eligible for the upgrade:
```

Availability of IPs::
+
Your VPC must have at least 25% free space, and 100 or more available IP addresses per subnet to be eligible for the upgrade.

Deprecated Availability Zones (AZs)::
+
VPCs using deprecated AZs aren't eligible for the upgrade.

Legacy VPNs::
+
VPCs with legacy VPNs configured aren't eligible for the upgrade.

Direct Connect or VPC Peering:: 
If your VPC has Direct Connect or VPC Peering configured, migrate to Transit Gateway or VPN before upgrading.

TLS 1.1 ciphers::
+
These ciphers can't be carried over from CloudHub to CloudHub 2.0:
+
* ECDHE-ECDSA-AES128-SHA1
* ECDHE-ECDSA-AES256-SHA1
* ECDHE-RSA-AES128-SHA1
* ECDHE-RSA-AES256-SHA1

== Upgrade your VPC via Anypoint Platform

. From *Anypoint Platform*, select *Runtime Manager* > *Upgrade*.
+
Alternatively, select *Runtime Manager* > *What's New* to read more about the VPC upgrade, and click *Continue to Upgrade* to start upgrading your VPCs.

> . Click *Upgrade Now*, under the *Upgrade Status* column for the VPC you want to upgrade.

Reviewer: ```suggestion
. Click *Upgrade Now* in the *Upgrade Status* column for the VPC to upgrade.
```

+
The *Upgrade to CloudHub 2.0* review page displays the configurations currently available in your VPC, as well as the configurations that are created during the upgrade.
+

> image::upgrade-ch2-review-conf.png[]

Reviewer: Can you add alt text here? ["Upgrade to CloudHub 2.0 review page showing available configurations"]
(or whatever the image is...)

+
. Click *Next* to continue.
. Provide a name for the new private space.
+ 
Optionally, you can specify xref:ps-gather-setup-info.adoc#cidr-block[reserved CIDRs] to connect to your new private space.
. Click *Create* to create the new private space.
+

> The *Upgrade to CloudHub 2.0* review page shows the progress of the VPC upgrade process, which takes approximately fifteen to thirty minutes.

Reviewer: ```suggestion
The *Upgrade to CloudHub 2.0* review page shows the progress of the VPC upgrade process, which takes about 15 to 30 minutes.
```

Reviewer: (use numerals for numbers higher than nine)

As soon as the upgrade process starts, the upgraded VPC is locked for any further updates. 

You can cancel the upgrade process to revert to your VPC at any time by clicking *Cancel Upgrade*. In that case, the newly created private space is deleted, and your VPC is unlocked.

> Once you have successfully provisioned the private space, the *Cancel Upgrade* button is no longer visible. However, you can still roll back the upgrade by deleting the private space.

Reviewer: ```suggestion
After you provision the private space, the *Cancel Upgrade* button no longer shows. However, you can still roll back the upgrade by deleting the private space.
```

> == Next Steps after VPC Upgrade

Reviewer: ```suggestion
== Next Steps After VPC Upgrade
```

> After you have completed the upgrade, the private space is fully provisioned and you can start deploying applications into it.

Reviewer: ```suggestion
After you complete the upgrade, the private space is fully provisioned and you can start deploying applications to it.
```

> You can start migrating your applications individually by creating new versions in the CloudHub 2.0 private space and then deleting the corresponding application in the CloudHub VPC. Once you’ve successfully migrated all your applications, make sure to delete your old VPC and any attached Dedicated Load Balancers on CloudHub.

Reviewer: ```suggestion
You can start migrating your applications individually by creating new versions in the CloudHub 2.0 private space and then deleting the corresponding application in the CloudHub VPC. Once you’ve successfully migrated all your applications, delete your old VPC and any attached dedicated load balancers on CloudHub.
```

> During your migration, you can continue to deploy and manage applications in your VPC to maintain business continuity. However, after the upgrade, your CloudHub Dedicated VPC is locked for any infrastructure changes to prevent configuration drift. This includes adjustments to environment or business group associations, network connections, and internal DNS. Make all changes in the newly created private space instead. Any configuration changes carried over to the CloudHub 2.0 private space are applied to the CloudHub VPC during your upgrade. 

Reviewer: ```suggestion
During your migration, you can continue to deploy and manage applications in your VPC to maintain business continuity. After the upgrade, your CloudHub dedicated VPC is locked for any changes to the infrastructure, including environment or business group associations, network connections, and internal DNS, to prevent configuration drift. Make all changes in the newly created private space instead. Any configuration changes carried over to the CloudHub 2.0 private space are applied to the CloudHub VPC during the upgrade. 
```

If your CloudHub VPC doesn't have any network connections and you create new ones in your new CloudHub 2.0 private space after the upgrade, these new connections aren’t reflected back in your CloudHub VPC.

> The firewall rules configured in your VPC get copied to the private space during the upgrade. After that, you can make additional changes to the firewall rules in both the VPC and the private space. To avoid configuration drift, apply any firewall rule changes in both places throughout your migration.

Reviewer: ```suggestion
The firewall rules configured in your VPC are copied to the private space. After that, you can make additional changes to the firewall rules in both the VPC and the private space. To avoid configuration drift, apply any firewall rule changes in both places throughout your migration.
```

== Entitlements for VPCs, Private Spaces, and Network Connections (VPNs/TGW)

> Once you complet the VPC upgrade and bring over any existing TGW and VPN network connections, both your old and new infrastructure are counted separately towards your usage. Therefore, if your organization is already using up all of your purchased quota, your usage can exceed the allowed quota until you move all your applications to CloudHub 2.0 and decommissioned your old VPC. 

Reviewer: ```suggestion
After you complete the VPC upgrade and bring over any existing TGW and VPN network connections, your old and new infrastructure count separately against your usage. Therefore, if your organization uses all purchased quota, your usage can exceed the allowed quota until you move all your applications to CloudHub 2.0 and decommission your old VPC. 
```

> Work with your account representative to plan for additional entitlements, and make sure you decommission your CloudHub infrastructure (VPCs and associated DLBs) within six months after the upgrade to avoid interruptions when trying to create or edit VPCs, private spaces, or network connections.

Reviewer: ```suggestion
Work with your account representative to plan for additional entitlements, and decommission your CloudHub infrastructure (VPCs and associated DLBs) within six months after the upgrade to avoid interruptions when trying to create or edit VPCs, private spaces, or network connections.
```

CloudHub VPC doesn't restrict outbound egress traffic, so removing the default firewall rule that allows this traffic can cause failures in your migrated private space.

== Upgrade your VPC via Anypoint Platform

. From *Anypoint Platform*, select *Runtime Manager* > *Upgrade*.
+
Alternatively, select *Runtime Manager* > *What's New* to read more about the VPC upgrade, and click *Continue to Upgrade* to start upgrading your VPCs.
. Click *Upgrade Now* in the *Upgrade Status* column for the VPC to upgrade.
+
The *Upgrade to CloudHub 2.0* review page displays the configurations currently available in your VPC, as well as the configurations that are created during the upgrade.
+
image::upgrade-ch2-review-conf.png[VPC upgrade review page showing available configurations.]
+
. Click *Next* to continue.
. Provide a name for the new private space.
+ 
Optionally, you can specify xref:ps-gather-setup-info.adoc#cidr-block[reserved CIDRs] to connect to your new private space.
. Click *Create* to create the new private space.
+
The *Upgrade to CloudHub 2.0* review page shows the progress of the VPC upgrade process, which takes about 15 to 30 minutes.

[NOTE]

> If you create a VPN to provision your CloudHub VPC, wait until the newly created VPN status shows as available before migrating your VPC to a CloudHub 2.0 private space. 

Reviewer 2: what do we mean by "create a VPN to provision your CloudHub VPC"?

Author: It refers to the VPN that customers need to configure to connect a newly created VPC to their network.

> = VPC Upgrade to Private Space 

Reviewer 3: CloudHub 1.0 VPC to CloudHub 2.0 Private Space Upgrade

> 

Reviewer 3: Overall missing the quotas piece - we need to clarify that they are using doubled entitlements for the duration of their upgrade - for both VPC & Network Connections. Therefore, to ensure they stay compliant, they need to work with their account team on entitlements and finish their upgrade in a timely manner.

//Overview

> CloudHub VPC upgrade is an out-of-the-box tool and guided UI experience for an infrastructure upgrade to accelerate the migration of a VPC into a Private Space.

Reviewer 3: I would avoid calling the feature "CloudHub VPC Upgrade". 

If you are looking to migrate your workloads from CloudHub 1.0 to CloudHub 2.0, you can kickstart your journey by first upgrading your CloudHub VPC to a CloudHub 2.0 Private Space using the in-product upgrade tool.

// Benefits

> This tool enables you to provision the infrastructure of a CloudHb 2.0 Private Space by cloning the existing configurations of eligible CloubHub VPCs, and carrying over any existing network connections, including TGWs and VPNs. 

Reviewer 3: The VPC to Private Space Upgrade tooling enables you to accelerate the creation of a CloudHub 2.0 Private Space by simply upgrading your existing CloudHub VPC. If you upgrade your CloudHub VPC into a Private Space instead of creating a new CloudHub 2.0 Private space, you can eliminate the time spent in acquiring new CIDR block, setting up new network connections, and configuring the private space from scratch. 

The tool enables you to provision the infrastructure of a CloudHub 2.0 Private Space by cloning the existing configurations of an eligible CloubHub VPC while leveraging the same CIDR block. If your VPC has existing TGW or VPN network connections, they will be carried over to the upgraded CloudHub 2.0 Private Space, and available for communication with any new applications deployed into the upgraded CloudHub 2.0 Private Space. 

Once your Private Space is created, proceed to incrementally migrate your applications from the CloudHub 1.0 VPC to the CloudHub 2.0 private space in a timely manner.

> At the same time, this UI experience reduces migration time and costs by eliminating a two- to-three-month toil to obtain all the necessary infrastructure and security approvals on new network connections and CIDR blocks. 

Reviewer 3: Delete - already incorporated above

Due to their architectural differences and distinct interfaces, CloudHub 2.0 doesn't support some features in the same way CloudHub does. Follow this eligibility check to confirm which VPCs are eligible for upgrade:

Availability of IPs::
+
Your VPC must have at least 25% free space, and 100 or more available IP addresses per subnet to be eligible for upgrade.

Deprecated Availability Zones (AZs)::
+
VPCs using deprecated AZs are not eligible for upgrade.

Legacy VPNs::
+
VPCs with legacy VPNs configured are not eligible for upgrade.

Direct Connect or VPC Peering:: 
If your VPC has Direct Connect or VPC Peering configured in their VPC, migrate to Transit Gateway before upgrading.

TLS 1.1 ciphers::
+
The following ciphers can't be carried over from CloudHub to CloudHub 2.0:
+
* ECDHE-ECDSA-AES128-SHA1
* ECDHE-ECDSA-AES256-SHA1
* ECDHE-RSA-AES128-SHA1
* ECDHE-RSA-AES256-SHA1

== Upgrade your VPC via Anypoint Platform

//Steps to upgrade

. From *Anypoint Platform*, select *Runtime Manager* > *Upgrade*.
+
Alternatively, select *Runtime Manager* > *What's New* to read about how the VPC upgrade works, and click *Continue to Upgrade* to start upgrading your VPCs.
+ 

> The *Upgrade to CloudHub 2.0* window lists all VPCs that are eligible for upgrade.

Reviewer 3: Confusing phrasing - the window also lists VPCs that are not eligible (not just the eligible ones).

. Click *Upgrade Now*, under the *Upgrade Status* column for the VPC you want to upgrade.
+

> The *Upgrade to CloudHub 2.0* review page shows what elements are already available in your VPC and what elements are created during the upgrade.

Reviewer 3: What is elements?

+
image::upgrade-ch2-review-conf.png[]
+
. Click *Next* to continue.
. Provide a name for the new private space. 
+ 

> Optionally, you can specify reserved CIDRs to connect to your new private space.

Reviewer 3: Need to hyperlink to reserved cidrs docs

. Click *Create* to create the new private space.
+
The *Upgrade to CloudHub 2.0* review page shows the progress of the VPC upgrade process, which takes approximately fifteen to thirty minutes.

As soon as the upgrade process starts, the upgraded VPC is locked for any further updates. 

You can cancel the upgrade process to revert to your VPC at any time by clicking *Cancel Upgrade*. In that case, the newly created private space is deleted, and your VPC is unlocked.

== Next Steps 

//after VPC Upgrade

> Once the upgrade process completes, the private space is fully provisioned and you can start deploying applications to it.

Reviewer 3: Add: You can begin migrating your applications one by one by creating new applications in the CloudHub 2.0 Private Space and deleting the equivalent application in the CloudHub 1.0 VPC. Once you are finished upgrading all applications, please delete your VPC & any attached Dedicated Load Balancers on CloudHub 1.0.

> You can continue to deployand manage applications into your VPC, but you can't change any VPC configurations. Make any infrastructure changes in the newly upgraded private space.

Reviewer 3: You can continue to deploy and manage applications in your VPC. However, after the upgrade, the CloudHub VPC will not allow any new infrastructure changes (such as environment or business group associations, changes to network connections and changes to Internal DNS). These changes will need to be made on the newly upgraded private space, going forward. These changes will also get applied to the CloudHub VPC for the duration of your upgrade. 
The firewall rules configured at the VPC will be copied into the Private Space at the time of upgrade. Thereafter, you will have the flexibility to make additional changes to the Firewall rules both the VPC as well as the Private Space. To avoid configuration drift, ensure that any Firewall rule changes are made in both places for the duration of your migration.

= CloudHub VPC to CloudHub 2.0 Private Space Upgrade  

//Overview
To migrate your workloads from CloudHub 1.0 to CloudHub 2.0, you have to start by setting up your CloudHub 2.0 infrastructure and network connections. One way to achieve this is by creating fresh CloudHub 2.0 Private Spaces with different configurations from what you had on CloudHub 1.0. However, if you prefer to keep your infrastructure configuration the same, you can use the in-product upgrade tool to easily upgrade your CloudHub dedicated VPC to a CloudHub 2.0 private space.
Once your new private space is ready, you can gradually migrate your applications from the CloudHub VPC to the CloudHub 2.0 private space. After all your applications have been successfully moved, you can decommission your old VPC.

// Benefits
This VPC to private space in-product upgrade tool helps you speed up the creation of a CloudHub 2.0 private space. Instead of starting from scratch, you can easily upgrade your existing CloudHub dedicated VPC. This process saves you time by eliminating the need to acquire new CIDR blocks, set up new network connections, and configure the private space from the ground up.

> The tool provisions the infrastructure for a CloudHub 2.0 private space by cloning the existing configurations of your eligible CloudHub dedicated VPC, using the same CIDR block. If your VPC has existing TGW or VPN network connections, they are seamlessly transferred to the new CloudHub 2.0 private space. This ensures that any new applications deployed into the new private space can communicate effectively.

Reviewer 4: We should also add this: The default Cloudhub certs will be configured as part of the private space's default certificates, and the custom domains from the DLB(s) of the migrated VPC will be created as part of the private space's custom TLS certificates.

+
Your VPC must have at least 25% free space, and 100 or more available IP addresses per subnet to be eligible for upgrade.

Deprecated Availability Zones (AZs)::
+
VPCs using deprecated AZs aren’t eligible for the upgrade.

Legacy VPNs::
+
VPCs with legacy VPNs configured aren’t eligible for the upgrade.

Direct Connect or VPC Peering:: 
If your VPC has Direct Connect or VPC Peering configured in their VPC, migrate to Transit Gateway or VPN before upgrading.

TLS 1.1 ciphers::
+
These ciphers can’t be carried over from CloudHub to CloudHub 2.0:
+
* ECDHE-ECDSA-AES128-SHA1
* ECDHE-ECDSA-AES256-SHA1
* ECDHE-RSA-AES128-SHA1
* ECDHE-RSA-AES256-SHA1

== Upgrade your VPC via Anypoint Platform

. From *Anypoint Platform*, select *Runtime Manager* > *Upgrade*.
+
Alternatively, select *Runtime Manager* > *What's New* to read about how the VPC upgrade works, and click *Continue to Upgrade* to start upgrading your VPCs.
. Click *Upgrade Now*, under the *Upgrade Status* column for the VPC you want to upgrade.
+
The *Upgrade to CloudHub 2.0* review page displays the configurations currently available in your VPC, as well as the configurations that are created during the upgrade.
+

> image::upgrade-ch2-review-conf.png[]

Reviewer 4: @luanamulesoft The link is taking me to 404. How can I see the image for this?

Author: @aishwarya-murali This image is the second file shown in this PR: cloudhub/modules/ROOT/assets/images/upgrade-ch2-review-conf.png

+
. Click *Next* to continue.
. Provide a name for the new private space.
+ 
Optionally, you can specify xref:ps-gather-setup-info.adoc#cidr-block[reserved CIDRs] to connect to your new private space.
. Click *Create* to create the new private space.
+
The *Upgrade to CloudHub 2.0* review page shows the progress of the VPC upgrade process, which takes approximately fifteen to thirty minutes.

As soon as the upgrade process starts, the upgraded VPC is locked for any further updates. 

> You can cancel the upgrade process to revert to your VPC at any time by clicking *Cancel Upgrade*. In that case, the newly created private space is deleted, and your VPC is unlocked.

Reviewer 4: Once the private space is successfully provisioned, the 'Cancel Upgrade' button is hidden. However, the user can still rollback by deleting the private space. We need to add this as well.
